File: app/src/main/java/com/xposed/spaceswitcher/HookMain.java
package com.xposed.spaceswitcher;

import de.robv.android.xposed.IXposedHookLoadPackage;
import de.robv.android.xposed.XposedBridge;
import de.robv.android.xposed.callbacks.XC_LoadPackage;

public class HookMain implements IXposedHookLoadPackage {
    @Override
    public void handleLoadPackage(XC_LoadPackage.LoadPackageParam lpparam) throws Throwable {
        try {
            if ("com.miui.securityspace".equals(lpparam.packageName)) {
                XposedBridge.log("[SpaceSwitch] Hook MIUI SecuritySpace");
                SpaceSwitchHook.hookSecuritySpace(lpparam);
            }
        } catch (Exception e) {
            XposedBridge.log("[SpaceSwitch] Error: " + e.getMessage());
        }
    }
}

File: app/src/main/java/com/xposed/spaceswitcher/HookMain.kt
package com.xposed.spaceswitcher

import de.robv.android.xposed.IXposedHookLoadPackage
import de.robv.android.xposed.XposedBridge
import de.robv.android.xposed.callbacks.XC_LoadPackage

class HookMain : IXposedHookLoadPackage {
    override fun handleLoadPackage(lpparam: XC_LoadPackage.LoadPackageParam) {
        try {
            if (lpparam.packageName == "com.miui.securityspace") {
                XposedBridge.log("[SpaceSwitch] Hook MIUI SecuritySpace")
                SpaceSwitchHook.hookSecuritySpace(lpparam)
            }
        } catch (e: Exception) {
            XposedBridge.log("[SpaceSwitch] Error: ${e.message}")
        }
    }
}
